packages/launchpad/cypress/e2e/integration/choose-a-browser.spec.ts

```diff
@@ -0,0 +1,280 @@
+import type { FoundBrowser } from '@packages/types/src'
+
+describe('Choose a Browser Page', () => {
+  // Walks through setup pages to get to browser selection
+  const stepThroughConfigPages = () => {
+    cy.get('h1').should('contain', 'Configuration Files')
+    cy.contains('Continue').click()
+    cy.get('h1').should('contain', 'Initializing Config...')
+    cy.contains('Next Step').click()
+  }
+
+  // Mocks server's browser detection with known list
+  const setupMockBrowsers = function () {
```

Comment:
Could accept a parameter which defaults to an empty list of browsers for the zero mock browser case and/or defaults to mockBrowsers list for the non-zero case

Reply:
We cannot access the outer scope from within the `withCtx` handler, which makes the logic hard to reuse/configure. There might be a way to interpolate values into these blocks dynamically, but it seems like it'd be a net-negative in terms of readability.

Reply:
Actually, I think I lied to you. You can pass additional args to withCtx that will be serialized into the second parameter:

```js
cy.withCtx(async (ctx, o) => {
  if (o.foo === 'bar') {
    // do the thing
  }
}, { foo: 'bar' })
```